### PR TITLE
feat: inherit the caller's model for async sub-agent launch and update

### PR DESCRIPTION
**Pull request by jfilipiuk**
## Summary

When the main run uses a free or cheap model but launches an async sub-agent, the sub-agent falls back to the config.yaml default model (e.g. `gemini-3-flash-preview`). On the `langgraph_server` backend the model-passthrough proxy runs inside the dev-server process, where `_ensure_config()` reports the server's config-default rather than the CLI's per-run choice - so the sub-agent silently bills the config-default and 402s when the account is out of credit. This PR makes an async sub-agent inherit the launching run's model at both launch and continuation.

### `EvoScientist/middleware/expert_async_subagent.py`

- `_caller_model_scope(runtime)` publishes the caller's per-run `(model, model_provider)` - read from `runtime.config` - to a ContextVar for the duration of a single `runs.create`. The model is taken from `runtime.config` (the same channel that carries `thread_id`) because `langgraph.config.get_config()` is unreliable from a tool's execution context.
- `start_async_task` (sync and async) wraps its `runs.create` in that scope.
- `update_async_task` is wrapped by `_build_expert_update_tool`, a thin delegator that sets the same scope and calls upstream's tool body unchanged, preserving its `multitask_strategy` and task-envelope semantics.

### `EvoScientist/llm/patches.py`

- The single `runs.create` merge point (`_merge_runs_config_kwargs`) layers the ContextVar over `_read_cfg_configurable()` as the highest-precedence model source. Callers with no per-run override still fall back to the config-default, so default-model behavior is unchanged. No behavior change on the in-process backend, where `_ensure_config()` already reports the live model.

## Tests

- `tests/test_model_passthrough_patch.py` - `_extract_caller_configurable` edge cases (none / no configurable / model-only / model+provider / bare-provider dropped); the caller's model beats a differing config-default in `_merge_runs_config_kwargs`; empty caller falls back to the config-default.
- `tests/test_expert_async_subagent.py` - start and update, sync and async, forward the caller's model over a differing config-default; the ContextVar resets after each call so nothing leaks to the next launch.

## Verification

- `uv run pytest tests/test_model_passthrough_patch.py tests/test_expert_async_subagent.py -q` - 39 pass.
- `uv run pytest tests/test_route_async_specs.py tests/test_async_subagent_swap.py tests/test_cli_async_runtime.py tests/test_expert_container_async.py tests/test_async_subagent_factory.py tests/test_async_watcher_middleware.py -q` - 62 pass.
- Pre-commit hooks: ruff check + ruff format pass on the changed files.

Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation / examples
- [ ] Test improvement
- [ ] Refactor (no behavior change)

Checklist

- [x] I have read the ../CONTRIBUTING.md
- [x] This targets core functionality used by the majority of users
- [x] I have added/updated tests where applicable
- [x] `uv run ruff check .` passes
- [x] `uv run pytest` passes

## Summary by CodeRabbit

* **New Features**
  * Expert asynchronous subagents now inherit the launching task’s model and provider settings.
  * Expert subagent launches automatically include their designated skill.
  * Previously unavailable expert subagents can now be resolved automatically when requested.
  * Synchronous and asynchronous follow-up tasks preserve caller-specific configuration.

* **Bug Fixes**
  * Prevented caller configuration from leaking between tool executions.
  * Ensured caller settings take precedence over default model configuration.
  * Improved consistency when launching and updating asynchronous subagent tasks.

### CODE REVIEW

File: EvoScientist/middleware/expert_async_subagent.py [hunk 1]
```diff
@@ -68,6 +69,72 @@
 _logger = logging.getLogger(__name__)
 
 
+@contextmanager
+def _caller_model_scope(runtime: ToolRuntime):
```
Comment (din0s) [inline EvoScientist/middleware/expert_async_subagent.py:88]: The docstring conflicts with `middleware/configurable_model.py` on whether `get_config()` is reliable or not. If what this function says is true, does it have any implications on the correctness of `ConfigurableModelMiddleware`?

Comment (jfilipiuk) [reply EvoScientist/middleware/expert_async_subagent.py:88]: Fixed, this docstring was an overstatement.

**Review by din0s — APPROVED** [review]

**Comment by jfilipiuk** [comment]
Rebased for conflict resolution and CI test.

**Review by X-iZhang — APPROVED** [review]
